guardian/service-catalogue
# feat(cloudquery): Make `OrgWideEc2` a singleton task

## What does this change, and why?
We have a few tasks that are starting to overlap on runs. This does not follow [CloudQuery’s advice](https://www.cloudquery.io/docs/advanced-topics/running-cloudquery-in-parallel#no-overlapping-syncs).

It’s difficult to predict how long a task will take. That is stopping the overlap via clever scheduling is tricky. This change implements [singleton](https://en.wikipedia.org/wiki/Singleton_pattern) tasks with the following behaviour:
  1. Task starts
  2. Check if another task in the same ECS family is currently running
  3. If yes, stop the task
  4. If no, continue to the CloudQuery container

This manifests as a `runAsSingleton` boolean property, which is defaulted to `false`.

In this change, the `OrgWideEc2` task becomes a singleton.

## How has it been verified?
- Deploy to CODE.
- Run the `OrgWideEc2` task twice (or more):

   ```sh
   npm -w cli start run-task -- --stage CODE --name OrgWideEc2
   ```
- Observe all but one getting to the to CloudQuery container.

   This can be done in the AWS console: 

   <img width="1018" alt="image" src="https://github.com/guardian/service-catalogue/assets/836140/6d54420b-13b2-4610-9142-5c4d5f199f8d">

   Or via [the logs](https://logs.gutools.co.uk/s/devx/goto/a8d0cb70-8539-11ee-b6f6-a11d207f34cc) (there should be a unique `ecs_task_arn` column).

---

Closes https://github.com/guardian/service-catalogue/pull/461.
